frontend/src/components/QueryBuilderV2/QueryV2/MetricsSelect/MetricsSelect.styles.scss

```diff
@@ -1,5 +1,10 @@
-.metrics-select-container {
+.metrics-source-select-container {
 	margin-bottom: 8px;
+	display: flex;
+	flex-direction: row;
+	align-items: flex-start;
+	gap: 8px;
+	width: 100%;
```

Comment:
Maybe give a fixed width to the select to avoid the shift?

https://github.com/user-attachments/assets/b2133b6e-0f66-4715-82b3-79105317d040